lib/model/content.dart

```diff
@@ -1054,8 +1055,11 @@ class _ZulipContentParser {
     return inlineParser.parseBlockInline(nodes);
   }
 
-  String? parseMathBlock(dom.Element element) {
-    return _parseMath(element, block: true);
+  BlockContentNode parseMathBlock(dom.Element element) {
+    final debugHtmlNode = kDebugMode ? element : null;
```

Comment:
Looks like we will end up with a different `debugHtmlNode` here, but that's a debug-mode-only behavior so it should be fine as a part of an [nfc] commit.

Reply:
Ah, yeah, would have been good for me to make that explicit.

There's also a related change that if it fails to parse, the UnimplementedNode will get the `span.katex-display` child instead of the `p` parent. I think that change is good (and it's definitely needed in the #1156 context, and in fact the current revision of #1156 effectively already makes that change) but it's not quite NFC.

I actually had that thought at one point as I was writing these changes; but I guess I forgot at the moment I was making that commit.